### PR TITLE
Improve canHandle method

### PR DESCRIPTION
**Pull request by Yoshani**
### Related Issues
- Issue https://github.com/wso2-enterprise/asgardeo-product/issues/21750

### Related PRs
- Related PR https://github.com/wso2/identity-apps/pull/5453

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.hypr/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/HyprAuthenticator.java [hunk 1]
```diff
@@ -164,7 +164,9 @@ public String getContextIdentifier(HttpServletRequest request) {
     public boolean canHandle(HttpServletRequest request) {
 
         // With HYPR if sessionDataKey is not received then the entire flow will break as it highly relied on it.
-        return request.getParameter(HYPR.SESSION_DATA_KEY) != null;
+        return StringUtils.isNotBlank(request.getParameter("authType")) &&
```
Comment (AfraHussaindeen) [inline components/org.wso2.carbon.identity.application.authenticator.hypr/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/HyprAuthenticator.java:167]:
```suggestion
        return StringUtils.isNotBlank(request.getParameter(HYPR.AUTH_TYPE)) &&
```

File: components/org.wso2.carbon.identity.application.authenticator.hypr/src/test/java/org/wso2/carbon/identity/application/authenticator/hypr/HyprAuthenticatorTest.java [hunk 2]
```diff
@@ -159,6 +160,8 @@ public void testCanHandle() {
 
         when(httpServletRequest.getParameter(HyprAuthenticatorConstants.HYPR.SESSION_DATA_KEY))
                 .thenReturn(sessionDataKey);
+        when(httpServletRequest.getParameter(HyprAuthenticatorConstants.HYPR.AUTH_TYPE))
+                .thenReturn(authType);
         Assert.assertTrue(hyprAuthenticator.canHandle(httpServletRequest));
```
Comment (AfraHussaindeen) [inline components/org.wso2.carbon.identity.application.authenticator.hypr/src/test/java/org/wso2/carbon/identity/application/authenticator/hypr/HyprAuthenticatorTest.java:166]: Add the test case related to the scenario where `authType` is null and `sessionDataKey` is a valid value.

```
        when(httpServletRequest.getParameter(HyprAuthenticatorConstants.HYPR.SESSION_DATA_KEY))
                .thenReturn(sessionDataKey);
        when(httpServletRequest.getParameter(HyprAuthenticatorConstants.HYPR.AUTH_TYPE))
                .thenReturn(null);
        Assert.assertFalse(hyprAuthenticator.canHandle(httpServletRequest));

```

**Review by AfraHussaindeen — APPROVED** [review]
